timepiece/crm/models.py

```diff
@@ -1008,6 +1008,11 @@ class Project(models.Model):
         unique=False,
         blank=True, # this field is required but is manually enforced in the code
         help_text="Auto-generated project code for tracking.")
+    ext_code = models.CharField(max_length=255,
+        verbose_name="External Project ID",
+        unique=False,
+        blank=True,
+        null=True)
```

Comment:
Just a note that you do not need to set null=True on a CharField or TextField.  Just blank=True because the Django ORM does not distinguish between an empty String and Null.